libmscore/system.cpp

```diff
@@ -642,11 +642,14 @@ void System::layout2()
             // if end staff not visible, try prev staff
             while (staffIdx1 <= staffIdx2 && !_staves[staffIdx2]->show())
                   --staffIdx2;
+            // it's user's choice to decide spanning to how many staves qualifies a bracket for being shown
+            // the decision is a score style "bracketMinStaves", referenced as BMS below
             // the bracket will be shown IF:
-            // it spans at least 2 visible staves (staffIdx1 < staffIdx2) OR
-            // it spans just one visible staff (staffIdx1 == staffIdx2) but it is required to do so
+            // it spans at least BMS visible staves OR
+            // it spans just less than BMS visible staff but it is required to do so
             // (the second case happens at least when the bracket is initially dropped)
-            bool notHidden = (staffIdx1 < staffIdx2) || (b->span() == 1 && staffIdx1 == staffIdx2);
+            int bms = score()->styleI(Sid::bracketMinStaves);
+            bool notHidden = (staffIdx1 < staffIdx2 + (2 - bms)) || (staffIdx1 == staffIdx2 + 1 - b->span());
```

Comment:
That `2` is the default as per libmscore/style.cpp, would be better to determine that programatically rather than having a 'magic number' here.
Also, while I certainly understand the change to the condition before the `||`, I don't for the the change after that.

Reply:
The change after `||` is necessary, because without it, if `bms` becomes very high, the brackets before the first system will disappear too.

Reply:
And `2` is not the default of `bms`, it's the same number, but here `2` is due to arithmetic reason. Here `2` is used for correct comparison between `staffIdx1` and staffIdx2`, whereas the `2` as the default value of `bms` is decided by the preference of majority users. So nothing magic about it, and changing it doesn't make any sense. ;-)

Reply:
OK, so a missunderstanding on my side